src/test/java/teammates/test/driver/GaeSimulation.java

```diff
@@ -240,12 +242,34 @@ private HttpServletRequest createWebRequest(String uri, String... parameters) {
     public static Map<String, Object> getEnvironmentAttributesWithApplicationHostname() {
         Map<String, Object> attributes = new HashMap<>();
         try {
+            if (!TestProperties.TEAMMATES_URL.contains("localhost")) {
+                attributes.put("com.google.appengine.runtime.default_version_hostname",
+                        new URL(TestProperties.TEAMMATES_URL).getAuthority().substring(10));
```

Comment:
Why `10`?

Reply:
To get rid of `6-6-0-dot-` part of hostname.

I think I will follow @xpdavid's suggestion and rather change the testing method to only check for params rather than the entire URL. So I will remove this code